src/chart/generateCategoricalChart.tsx

```diff
@@ -842,22 +844,34 @@ export interface CategoricalChartProps {
 }
 
 type AxisObj = {
-  xAxis?: BaseAxisProps;
+  xAxis?: XAxisProps;
   xAxisTicks?: Array<TickItem>;
 
-  yAxis?: BaseAxisProps;
+  yAxis?: YAxisProps;
   yAxisTicks?: Array<TickItem>;
 
-  zAxis?: BaseAxisProps;
+  zAxis?: ZAxisProps;
   zAxisTicks?: Array<TickItem>;
 
-  angleAxis?: BaseAxisProps;
+  angleAxis?: AngleAxisProps;
   angleAxisTicks?: Array<TickItem>;
 
-  radiusAxis?: BaseAxisProps;
+  radiusAxis?: RadiusAxisProps;
   radiusAxisTicks?: Array<TickItem>;
 };
 
+// Determine the size of the axis, used for calculation of relative bar sizes
+const getCartesianAxisSize = (axisObj: AxisObj, axisName: 'xAxis' | 'yAxis' | 'angleAxis' | 'radiusAxis') => {
```

Comment:
nit: `getCartesianAxisSize` seems generic enough that one could accidentally use this function in the future for something else without realizing how specific it is to calculating `barSize`. 

We should either name this slightly more specific (the comments are specific, so the function can be named specific too) or more generically and add JSDoc that helps future us understand what this could possibly be used for

Reply:
Yeah I think this can be generic, I'll add JSDoc for it. Maybe there's a better word instead of `size`? Like if we think CSS flex-box, the size is kind of the length of the "main axis"?